remotes/docker/resolver.go

```diff
@@ -291,10 +291,14 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 			resp.Body.Close() // don't care about body contents.
 
 			if resp.StatusCode > 299 {
-				if resp.StatusCode == http.StatusNotFound {
-					continue
+				if resp.StatusCode > 399 {
+					// Set lastErr when encountering the first non-404 status code.
+					if lastErr == nil && resp.StatusCode != http.StatusNotFound {
```

Comment:
nit: `lastErr` doesn's seems to be the correct name as this actually indicates the first error :thinking:

Reply:
Right, but that seems to already be the case at:

- https://github.com/containerd/containerd/blob/7b9a4ec09280962a801077c15fdf5ff8269c0796/remotes/docker/resolver.go#L286
- https://github.com/containerd/containerd/blob/7b9a4ec09280962a801077c15fdf5ff8269c0796/remotes/docker/resolver.go#L364
- https://github.com/containerd/containerd/blob/7b9a4ec09280962a801077c15fdf5ff8269c0796/remotes/docker/resolver.go#L381

Should I rename it?

Reply:
Done